test/instance_unittest.cc

```diff
@@ -0,0 +1,77 @@
+#include "../src/configuration.h"
+#include "../src/environment.h"
+#include "../src/instance.h"
+#include "../src/resource.h"
+#include "../src/updater.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+namespace {
+
+class InstanceTest : public ::testing::Test {
+ protected:
+  const MetadataStore::Metadata&& GetResourceMetadata(
+      const MetadataUpdater::ResourceMetadata* rm) {
+    return std::move(rm->metadata);
+  }
+
+  MonitoredResource GetMonitoredResource(
```

Comment:
Ditto here: the `ResourceMetadata` object already owns that value — just get a const reference to look at (and avoid a copy):
```cpp
  const MonitoredResource& GetMonitoredResource(
      const MetadataUpdater::ResourceMetadata& rm) {
    return rm->resource;
  }
```
Also in `GetResourceIds`.

Reply:
I don't see a const reference being returned. Also in `GetResourceIds`.

Reply:
Done.